src/go/rpk/pkg/net/hostport.go

```diff
@@ -138,6 +146,10 @@ func isDomain(d string) bool {
 // underscores in the middle.
 var labelRe = regexp.MustCompile(`^[a-zA-Z0-9](?:[a-zA-Z0-9_-]*[a-zA-Z0-9])?$`)
 
+// top level domain can be alphanumeric, include dashes or undescores
+// but not only numbers.
```

Comment:
https://stackoverflow.com/questions/9071279/number-in-the-top-level-domain

I think we can change this to:
```
^(?:[a-zA-z]+|xn--[a-zA-Z0-9]+)$`
```
entirely alphabetic, OR xn-- and alphanumeric

Technically the entirely alphabetic is a bit strong of a restriction, but no non-IDN TLD uses a number.

Reply:
Changed and added a new unit test for these cases.